src/opendr/perception/object_detection_2d/utils/class_filter_wrapper.py

```diff
@@ -0,0 +1,137 @@
+                "N/A", "handbag", "tie", "suitcase", "frisbee", "skis", "snowboard", "sports ball", "kite",
+                "baseball bat", "baseball glove", "skateboard", "surfboard", "tennis racket", "bottle", "N/A",
+                "wine glass", "cup", "fork", "knife", "spoon", "bowl", "banana", "apple", "sandwich", "orange",
+                "broccoli", "carrot", "hot dog", "pizza", "donut", "cake", "chair", "couch", "potted plant", "bed",
+                "N/A", "dining table", "N/A", "N/A", "toilet", "N/A", "tv", "laptop", "mouse", "remote", "keyboard",
+                "cell phone", "microwave", "oven", "toaster", "sink", "refrigerator", "N/A", "book", "clock", "vase",
+                "scissors", "teddy bear", "hair drier", "toothbrush",
+            ]
+            self.classes = coco_classes
+
+        # Verify that allowed classes are in the detector's class list
+        invalid_classes = [cls for cls in self.allowed_classes if cls not in self.classes]
+        if invalid_classes:
+            raise ValueError(
+                f"The following classes are not detected by this detector: {', '.join(invalid_classes)}")
+
+    def infer(self, img=None, threshold=None, keep_size=None, input=None, conf_threshold=None, iou_threshold=None,
+              nms_max_num=None, nms_threshold=None, scales=None, mask_thresh=None, size=None, custom_nms=None,
+              nms_thresh=None, nms_topk=None, post_nms=None, extract_maps=None):
+
+        # match variable names
+        if isinstance(self.learner, NanodetLearner):
+            img = input
+
+        if img is None:
+            raise ValueError(
+                "An image input is required. Please provide a valid image.")
+
+        if isinstance(self.learner, CenterNetDetectorLearner):
+            if threshold is None:
+                threshold = 0.2
+            if keep_size is None:
+                keep_size = True
+            boxes = self.learner.infer(img, threshold=threshold, keep_size=keep_size)
+
+        elif isinstance(self.learner, YOLOv3DetectorLearner):
+            if threshold is None:
+                threshold = 0.1
+            if keep_size is None:
+                keep_size = True
+            boxes = self.learner.infer(img, threshold=threshold, keep_size=keep_size)
+
+        elif isinstance(self.learner, YOLOv5DetectorLearner):
+            if size is None:
+                size = 640
+            boxes = self.learner.infer(img, size)
+
+        elif isinstance(self.learner, DetrLearner):
+            boxes = self.learner.infer(img)
+
+        elif isinstance(self.learner, NanodetLearner):
+            if conf_threshold is None:
+                conf_threshold = 0.35
+            if iou_threshold is None:
+                iou_threshold = 0.6
+            if nms_max_num is None:
+                nms_max_num = 100
+            boxes = self.learner.infer(input, conf_threshold, iou_threshold, nms_max_num)
```

Comment:
If `input` is provided it will be assigned to `img` anyway so we can use `img` here instead.

```suggestion
            boxes = self.learner.infer(img, conf_threshold, iou_threshold, nms_max_num)
```